src/operator/activation.cc

```diff
@@ -29,12 +29,30 @@
 #include "./mkl/mkl_memory-inl.h"
 #include "./mkl/mkl_relu-inl.h"
 #endif  // MXNET_USE_MKL2017
+#if MXNET_USE_MKLDNN == 1
+#include <mkl_memory.h>
+#include "./mkl/mkldnn_memory-inl.h"
+#include "./mkl/mkldnn_relu-inl.h"
+#endif  // MXNET_USE_MKLDNN
 
 namespace mxnet {
 namespace op {
 template<>
 Operator *CreateOp<cpu>(ActivationParam param, int dtype, const TShape& dshape) {
   Operator *op = NULL;
+#if MXNET_USE_MKLDNN == 1
+  if (param.act_type == activation::kReLU) {
+    switch (dtype) {
+    case mshadow::kFloat32:
+    case mshadow::kInt8:
+    case mshadow::kInt32:
+    case mshadow::kUint8:
+      return new MKLDNNReluOp<cpu, float>();
```

Comment:
Should it be allowed to turn on both MKLDNN and MKL2017? MKLRelu below supports double which isn't supported by mkldnn here.

Reply:
This is missing. I will add double for MKLDNN.

Reply:
Either one of MKLML(2017) or MKLDNN should be used. They cannot be used together for now.